OpenRA.Mods.Common/Activities/UnloadCargo.cs

```diff
@@ -111,17 +111,17 @@ public override bool Tick(Actor self)
 				}
 
 				cargo.Unload(self);
+
+				var pos = actor.Trait<IPositionable>();
+				pos.SetVisualPosition(actor, spawn);
+				var move = actor.Trait<IMove>();
+				actor.QueueActivity(false, move.MoveIntoWorld(actor, exitSubCell.Value.First, exitSubCell.Value.Second));
+
 				self.World.AddFrameEndTask(w =>
 				{
 					if (actor.Disposed)
 						return;
 
-					var move = actor.Trait<IMove>();
-					var pos = actor.Trait<IPositionable>();
-
-					actor.CancelActivity();
```

Comment:
Why remove this `CancelActivity`?

Reply:
This is already taken care of by `queued = false` in `actor.QueueActivity`

Reply:
Ah yes, I see.